### ISSUE TITLE
[BUG] In Rancher 2.7, GitHub teams are not selectable when using GitHub as Auth Provider

### ISSUE THREAD

**Issue by Napsty** [rancher/rancher#41980]
**Rancher Server Setup**
- Rancher version: 2.7.5
- Installation option (Docker install/Helm Chart): Docker
- Proxy/Cert Details: N/A

**Information about the Cluster**
- Kubernetes version: 
- Cluster Type (Local/Downstream): local cluster (custom)

**User Information**
- What is the role of the user logged in? Admin

**Describe the bug**
Selected GitHub as Auth Provider and successfully connected the OAuth application. Able to log in with a GitHub user from the organization. Tried to Assign Global Roles to a GitHub team, which has worked in the past (Rancher 2.5). GitHub teams are not showing up in the Rancher 2.7 UI. 

**To Reproduce**
- Start new Rancher 2.7.5 using Single Docker Installation
- Users&Authentication -> Auth Provider -> Set Up Github as Auth Provider
- Users&Authentication -> Groups -> Assign Global Roles To Group -> Select Member: No teams from Github (from our organization) are shown.

**Result**
No groups from Github (from our organization) are shown. Looks like only organizations can be selected, not teams within an organization.

**Expected Result**
GitHub teams from within the organization are shown and can be selected (as it was possible in Rancher 2.5).

**Screenshots**

Rancher 2.5:

![image](https://github.com/rancher/rancher/assets/3702203/091967b2-d539-443e-850a-8cd3f7b960cf)

Rancher 2.7:

![image](https://github.com/rancher/rancher/assets/3702203/f6ea477c-60d8-4bc1-b093-26a4acf7797f)

![image](https://github.com/rancher/rancher/assets/3702203/b45937a2-68e2-4c7c-bbdd-c3c1675a46f8)

**Comment by nickwsuse** [comment]
@Napsty I wasn't able to reproduce this on v2.7.5 - I see GitHub teams appearing on the Global Roles assignment as well as in the configuration of the GitHub auth provider. If you can provide any additional details to help reproduce this, please let me know and I can take a look.

Global Roles:
![image](https://github.com/rancher/rancher/assets/107147272/96e7360a-02bc-4d7f-a9a2-02a62e8a242e)

GH Auth Provider Config:
![image](https://github.com/rancher/rancher/assets/107147272/067a659e-7d14-41f0-8693-c83ec8b4340d)

**Comment by Napsty** [comment]
@nickwsuse Can you describe step-by-step how you added GitHub as Auth Provider? Maybe it's a permission thingy with the GitHub user used for Oauth2 application? However in GitHub itself, the user can see the teams.

**Comment by nickwsuse** [comment]
@Napsty I did the following steps on both an HA and on a single docker install of Rancher v2.7.5

1. Install Rancher v2.7.5
2. Log into Rancher
3. Click the hamburger menu in the upper left corner
4. Click Users & Authentication
![image](https://github.com/rancher/rancher/assets/107147272/e3aac6b1-bb75-4552-bc58-c4f92bf7553f)
5. Select the Github (OAuth) option
![image](https://github.com/rancher/rancher/assets/107147272/8f8629f9-24b7-4dfa-a1d8-8dc58e31259b)
6. I followed all the steps listed on this page
![image](https://github.com/rancher/rancher/assets/107147272/590309c2-f83d-4229-a8ec-72bc2653dcfc)
7. Once the OAuth app has been created on GH, I generated a Client Secret
8. I then added the Client ID and Client Secret at the bottom of the page in the screenshot above
9. Click Enable at the bottom right of the page
10. On the popup, I left it as it was (Rancher org enabled, other orgs I'm part of disabled)
11. Click the Authorize button

Once the authorization was complete, I was able to click into the search field for the GH auth provider sections and the Global Roles section and got the suggestions like I did in my previous screenshots. 

I'm not sure why you aren't getting the suggestions without typing, but I would guess it has something to do with how many orgs/teams you have access to, but again that's just a guess.

**Comment by nickwsuse** [comment]
@Napsty No, I did not enable Device Flow in the Github app. My initial thought was that the GH orgs that are selected may not allow teams to be viewed/accessible. [Here's a doc](https://docs.github.com/en/organizations/organizing-members-into-teams/changing-team-visibility) to check/update that just in case. I know you also mentioned in the original ticket body that it worked previously in Rancher v2.5, so something with the orgs themselves in GH is my current guess as to why you aren't able to view teams in v2.7.5

**Comment by Napsty** [comment]
I checked the teams settings on GitHub and they are set to Visible (team visibility). I suppose that's the same setting as your team(s)?

**Comment by nickwsuse** [comment]
Correct, the two teams that I'm part of are both visible in the Rancher org. The only other thing I could think of that could be causing issues would be an upgrade path if your current Rancher setup is the result of an upgrade from a previous version.

If it is, let me know what version(s) you upgraded from and I can try to follow the same path to see if I can reproduce it that way. If it still didn't work for you even on a fresh v2.7.5 install then I'm out of ideas, but could try to have someone else in QA try to reproduce it in case I'm just really lucky or something.

**Comment by Napsty** [comment]
Thank you for sticking with me @nickwsuse , much appreciated. No, it was a clean and new installation (Single Docker installation), no upgrade. 
I will do another new installation in another environment to see if I can see a difference.

**Comment by Napsty** [comment]
@nickwsuse I created a new Ubuntu 22.04 VM, installed docker.io, launched Rancher from the cli:

```
root@ubuntu2204:~# docker run -d --restart=unless-stopped --privileged -p 80:80 -p 443:443 rancher/rancher:v2.7.5
```

Then once logged in as admin (using the bootstrap password), added the Github authentication provider. 
Here I noticed something in the PopUp:

![image](https://github.com/rancher/rancher/assets/3702203/935503e9-c9c5-419a-828e-8d1cf9691374)

As you can see, the GitHub user I am using is not maintainer/admin of the Organization so I needed to send the request. 
I continued and created the OAuth Application in GitHub, copied the Client ID and the Secret, entered it Rancher and enabled GitHub Auth Provider.

At this point, again the same issue: No GitHub teams are showing up in the drop down menu when trying to assign a "Member". 

A couple of minutes later I got an e-mail, that my request was approved by the Github organization admin. 

I tried it again in Rancher, but still no teams in the drop down list.

Edited the Github Authentication Provider and created a new secret in Github, entered this new secret in Rancher's config and saved. 

Heureka! Now I was able to select GitHub teams from the drop down list!

Long story short: The **GitHub teams only show up when the OAuth application is already granted to access the organization**. Even if the request is granted later on, this is not sufficient, you need to create a new secret and use this in the GitHub Authentication Provider settings in Rancher.

Maybe someone could review/verify this finding.
